### PR TITLE
[state] Remove receipt checking after runAction

### LINKED ISSUE #3555: [workingset] Investigate `nil` receipt
### What would you like to be added
Investigate when the receipt is returned with `nil` value
```
receipt, err := ws.runAction(ctx, elp)
if err != nil {
	return nil, errors.Wrap(err, "error when run action")
}
if receipt != nil {
	receipts = append(receipts, receipt)
}
```

### Why is this needed
https://github.com/iotexproject/iotex-core/blob/master/state/factory/workingset.go#L183

### How important you think this is for IoTeX
- [] must have
- [x] should have
- [] nice to have

### Additional information
Let us know any background or context that would help us better understand the request (for example the particular use-case that prompted this request)

### PR DESCRIPTION
**Pull request by Liuhaai**
# Description
fix #3555 

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by CoderZhi — COMMENTED** [review]
what about https://github.com/iotexproject/iotex-core/pull/3780/files#diff-5168c619aa2e8281d3f34e0423560a0a1fda52f9fd1a49e57ba8390f0b3bfbd3R161

File: state/factory/factory_test.go [hunk 1]
```diff
@@ -729,9 +729,15 @@ func TestNonce(t *testing.T) {
 	cfg := DefaultConfig
 	db1, err := db.CreateKVStore(db.DefaultConfig, testTriePath)
 	require.NoError(t, err)
-	sf, err := NewFactory(cfg, db1, SkipBlockValidationOption())
+
+	reg := protocol.NewRegistry()
+	acc := account.NewProtocol(rewarding.DepositGas)
+	err = acc.Register(reg)
+	require.NoError(t, err)
```
Comment (CoderZhi) [inline state/factory/factory_test.go:736]: `require.NoError(t, acc.Register(reg))`

**Review by CoderZhi — APPROVED** [review]

File: state/factory/workingset.go [hunk 2]
```diff
@@ -478,11 +475,9 @@ func (ws *workingSet) pickAndRunActions(
...
 				return nil, errors.Wrapf(err, "Failed to update state changes for selp %x", nextActionHash)
 			}
-			if receipt != nil {
-				blkCtx.GasLimit -= receipt.GasConsumed
-				ctxWithBlockContext = protocol.WithBlockCtx(ctx, blkCtx)
-				receipts = append(receipts, receipt)
-			}
+			blkCtx.GasLimit -= receipt.GasConsumed
+			ctxWithBlockContext = protocol.WithBlockCtx(ctx, blkCtx)
+			receipts = append(receipts, receipt)
```
Comment (dustinxie) [inline state/factory/workingset.go:480]: remove empty line

File: state/factory/workingset.go [hunk 3]
```diff
@@ -160,16 +158,15 @@ func (ws *workingSet) runAction(
 	// Handle action
 	reg, ok := protocol.GetRegistry(ctx)
 	if !ok {
-		return nil, nil
+		return nil, errors.New("protocol is empty")
```
Comment (dustinxie) [inline state/factory/workingset.go:161]: could this potentially affect fullsync of past blocks?

**Review by dustinxie — APPROVED** [review]

**Pull request merged**